### DISCUSSION TITLE
Allowing redirect after login to external url

### DISCUSSION THREAD

**Discussion by loulou2852** [directus/directus#18811] [Feature Requests]
### Summary

Hello,

I try to simplify the process of login with Directus for an external website.
The purpose is to simplify the login part of the website.

I've got a project with a Directus instance: directus.mysite.com
And an external website located at mysite.com

I've assigned REFRESH_TOKEN_COOKIE_DOMAIN parameter with mysite.com

I want to be able to log in to the Directus instance and then be redirected to the main website.

Actually, the redirect parameter is not working outside the Directus instance due to potential security risks, I think.

With this URL: https://directus.mysite.com/admin/login?redirect=https%3A%2F%2Fmysite.com (ie https://mysite.com)
I've got this result: directus.mysite.com/admin/https://mysite.com
Instead of going to https://mysite.com

### Motivation

With this feature implemented, it will be easier to access Directus data with an external website.

This allows me to let Directus handle the login process 100%.
When it redirects, I've got the access token, and with the SDK script, it handles the refresh token automatically.

This lets focus on the external app/website, instead of authentication.

### Detailed Design

Possible solution to solve this issue:

getting this URL: https://directus.mysite.com/admin/login?redirect=https%3A%2F%2Fmysite.com

Checking if the redirect variable is an external domain.
Checking if the domain matches the REFRESH_TOKEN_COOKIE_DOMAIN variable.

If both are OK, then redirect to this external website.

### Must Have:
- improve the redirect parameter with possblie redirect to external website

### Should Have:
- Compare with REFRESH_TOKEN_COOKIE_DOMAIN for security reason

### Could Have:
- Nothing

### Won't Have:
- Nothing

### Drawbacks

With security protection with REFRESH_TOKEN_COOKIE_DOMAIN variable, there are no drawbacks.

### Alternatives

Create another variable in the URI which pops a button or a modal that invites the user to redirect manually to the URL.

### Adoption Strategy

If implemented, it will give more freedom to existing users and no breaking changes will be required.

**Comment by br41nslug** [comment]
I dont see an issue with redirecting itself however this redirect cannot provide an access_token or share cookies across domains as that would be a security issue. 

Im curious why run this flow through the directus login page with redirects instead of using the SDK to login directly? or even setting up an SSO provider?

**Comment by loulou2852** [comment]
![image](https://github.com/directus/directus/assets/4818634/cb18254c-047d-4296-a415-4db925a72715)
As for this upcoming feature, having the sign-up and sign-in page at the same place will be accurate in some examples.

I didn't know this type of redirect can't provide an access_token. But if I try to log-in with the Directus backend and then go to the external website manually, I'm logged-in too on the custom one. 

That why I thought this redirect will automatically do the process for me.

I was planning to use the SDK for that, but it needs those 3 features to develop with the UI -> login, logout, modify contact info
and with the Directus backend, everything is available out of the box.

I didn't try yet the SSO one, I'll use it specifically on another project.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
